src/schema/v2/me/updateCollectionsMutation.ts

```diff
@@ -0,0 +1,105 @@
+import {
+  GraphQLNonNull,
+  GraphQLList,
+  GraphQLString,
+  GraphQLBoolean,
+  GraphQLInputObjectType,
+  GraphQLObjectType,
+  GraphQLUnionType,
+} from "graphql"
+import { mutationWithClientMutationId } from "graphql-relay"
+import { ResolverContext } from "types/graphql"
+import { CollectionType } from "./collection"
+import {
+  GravityMutationErrorType,
+  formatGravityError,
+} from "lib/gravityErrorHandler"
+import { snakeCase } from "lodash"
+
+const SuccessType = new GraphQLObjectType<any, ResolverContext>({
+  name: "UpdateCollectionsSuccess",
+  isTypeOf: (data) => data.id,
+  fields: () => ({
+    collection: {
+      type: CollectionType,
+      resolve: (response) => response,
+    },
+  }),
+})
+
+const FailureType = new GraphQLObjectType<any, ResolverContext>({
+  name: "UpdateCollectionsFailure",
+  isTypeOf: (data) => data._type === "GravityMutationError",
+  fields: () => ({
+    mutationError: {
+      type: GravityMutationErrorType,
+      resolve: (err) => err,
+    },
+  }),
+})
+
+const ResponseOrErrorType = new GraphQLUnionType({
+  name: "UpdateCollectionsResponseOrError",
+  types: [SuccessType, FailureType],
+})
+
+export const CollectionsInputType = new GraphQLInputObjectType({
+  name: "CollectionsInput",
+  fields: {
+    id: {
+      type: new GraphQLNonNull(GraphQLString),
+    },
+    shareableWithPartners: {
+      type: new GraphQLNonNull(GraphQLBoolean),
+    },
+  },
+})
+
+export const updateCollectionsMutation = mutationWithClientMutationId<
+  any,
+  any | null,
+  ResolverContext
+>({
+  name: "updateCollectionsMutation",
+  description: "Updates the user's collections in batch.",
+  inputFields: {
+    attributes: {
+      type: new GraphQLNonNull(
+        new GraphQLList(new GraphQLNonNull(CollectionsInputType))
+      ),
+    },
+  },
+  outputFields: {
+    collectionsOrErrors: {
+      type: new GraphQLNonNull(
+        new GraphQLList(new GraphQLNonNull(ResponseOrErrorType))
+      ),
+      resolve: (result) => result,
+    },
+  },
+  mutateAndGetPayload: async (args, { meUpdateCollectionsLoader }) => {
+    if (!meUpdateCollectionsLoader) {
+      throw new Error("You need to be signed in to perform this action")
+    }
+
+    const gravityPayload = args.attributes.map((collection) => {
+      return Object.keys(collection).reduce(
+        (acc, key) => ({ ...acc, [snakeCase(key)]: collection[key] }),
+        {}
+      )
+    })
+
+    try {
+      return await meUpdateCollectionsLoader({
+        attributes: JSON.stringify(gravityPayload),
+      })
+    } catch (error) {
+      const formattedErr = formatGravityError(error)
+      if (formattedErr) {
+        return [{ ...formattedErr, _type: "GravityMutationError" }]
```

Comment:
The corresponding Gravity batch endpoint returns

- An array of updated collections when succeeded
- A single [_object_](https://github.com/artsy/gravity/blob/fa66526dafd211963f9db0fe677e4518c4e610d4/spec/api/v1/me_collections_endpoint_spec.rb#L115-L116) of an error when failed.

Here in MP we turn the error response type into an array as well. I guess it's fine (or even more flexible in the future a Gravity batch endpoint can return multiple errors).